code/modules/mob/living/silicon/robot/robot_mob.dm

```diff
@@ -550,6 +550,10 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 		weather_immunities -= "lava"
 	armor = getArmor(arglist(initial(armor)))
 
+	for(var/obj/item/borg/upgrade/U in contents)
+		QDEL_NULL(U)
```

Comment:
are the borg upgrades tracked in any other verb/list other than `contents`?

Reply:
They're not tracked anywhere else that I can see. I'm honestly not entirely sure where the upgrades are added to `contents` in the first place, but in testing this fixed the issue I wanted to fix, where after being reset you can't apply a VTEC upgrade again but you lose the speed.

Reply:
I just looked, why the fuck are borg upgrades only tracked in the contents, ugh. just ignore my comment, this is currently the "right" way to go about this san an entire refactor of this.